src/vs/workbench/contrib/search/browser/searchActions.ts

```diff
@@ -441,22 +441,156 @@ export abstract class AbstractSearchAndReplaceAction extends Action {
 	}
 }
 
+class ReplaceActionRunner {
+	constructor(
+		private viewer: WorkbenchObjectTree<RenderableMatch>,
+		private viewlet: SearchView | undefined,
+		private getElementToFocusAfterRemoved: (viewer: WorkbenchObjectTree<RenderableMatch>, elementToBeRemoved: RenderableMatch) => RenderableMatch,
+		private getPreviousElementAfterRemoved: (viewer: WorkbenchObjectTree<RenderableMatch>, element: RenderableMatch) => RenderableMatch,
+		// Services
+		@IReplaceService private readonly replaceService: IReplaceService,
+		@IEditorService private readonly editorService: IEditorService,
+		@IConfigurationService private readonly configurationService: IConfigurationService,
+		@IUriIdentityService private readonly uriIdentityService: IUriIdentityService
+	) { }
+
+	async performReplace(element: FolderMatch | FileMatch | Match): Promise<any> {
+		// since multiple elements can be selected, we need to check the type of the FolderMatch/FileMatch/Match before we perform the replace.
+		const elementsToReplace = getElementsToOperateOnInfo(this.viewer, element, this.configurationService.getValue<ISearchConfigurationProperties>('search'));
+
+		await Promise.all(elementsToReplace.map(async (elem) => {
+			const parent = elem.parent();
+
+			if ((parent instanceof FolderMatch || parent instanceof FileMatch) && arrayContainsElementOrParent(parent, elementsToReplace)) {
+				// skip any children who have parents in the array
+				return;
+			}
+
+			if (elem instanceof FileMatch) {
+				elem.parent().replace(elem);
+			} else if (elem instanceof Match) {
+				elem.parent().replace(elem);
+			} else if (elem instanceof FolderMatch) {
+				await elem.replaceAll();
+			}
+		}));
+
+		const currentBottomFocusElement = elementsToReplace[elementsToReplace.length - 1];
+
+		if (currentBottomFocusElement instanceof Match) {
```

Comment:
Does the order (for sorting) only matter because of this?

Reply:
pretty much, yes

Reply:
is there an easier way to get the last element? It would be different based on the sort order that the user specifies in their config